Tests/MLXTests/ModuleTests.swift

```diff
@@ -770,4 +791,97 @@ class ModuleTests: XCTestCase {
         XCTAssertTrue(pm.mlp.0 is QuantizedLinear)
         XCTAssertTrue(pm.mlp.2 is QuantizedLinear)
     }
+
+    func testCompositeLayer() throws {
+        // a test of making a LoRA layer as a composite -- it adapts
+        // another layer but mixes their properties.  this is not
+        // necessarily the way to implement LoRA (you can also use subclasses
+        // like Linear/QuantizedLinear) but this verifies that it can
+        // be written this way.
+
+        class LoRA: Module, UnaryLayer {
```

Comment:
A simple implementation of LoRA using composition, see https://github.com/ml-explore/mlx-swift-examples/pull/167.  This isn't meant as a reference implementation (though it may grow into that), but rather a test of the subclassing to make sure we have it all covered.